[CI] RerankIT class failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #12107 / ubuntu-2204_platform-support-unix](https://gradle-enterprise.elastic.co/s/i4zmqqglf3xog)
- [elasticsearch-periodic-platform-support #12107 / ubuntu-2204_platform-support-unix](https://gradle-enterprise.elastic.co/s/afdqn5gl4lwy2)
- [elasticsearch-periodic #12113 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/gpkdglx6mxzt4)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.plugin.RerankIT.testRerankRowLimitEnforcement" -Dtests.seed=DD08F7617D3EF6E9 -Dtests.locale=az -Dtests.timezone=America/Regina -Druntime.java=26
```

**Applicable branches:**
9.3

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.plugin.RerankIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('org.elasticsearch.xpack.esql.plugin.RerankIT'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: timeout
```

**Issue Reasons:**
- [9.3] 1 consecutive failures in class org.elasticsearch.xpack.esql.plugin.RerankIT
- [9.3] 2 consecutive failures in step ubuntu-2204_platform-support-unix
- [9.3] 1 consecutive failures in pipeline elasticsearch-periodic-platform-support
- [9.3] 3 failures in class org.elasticsearch.xpack.esql.plugin.RerankIT (1.6% fail rate in 183 executions)
- [9.3] 2 failures in step ubuntu-2204_platform-support-unix (40.0% fail rate in 5 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/AbstractEsqlIntegTestCase.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/AbstractEsqlIntegTestCase.java
@@ -38,7 +38,6 @@ import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Iterator;
 import java.util.List;
-import java.util.concurrent.TimeUnit;
 
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
 import static org.elasticsearch.xpack.esql.EsqlTestUtils.getValuesList;
@@ -48,6 +47,9 @@ import static org.hamcrest.Matchers.equalTo;
 
 @TestLogging(value = "org.elasticsearch.xpack.esql.session:DEBUG", reason = "to better understand planning")
 public abstract class AbstractEsqlIntegTestCase extends ESIntegTestCase {
+
+    protected static final TimeValue DEFAULT_REQUEST_TIMEOUT = TimeValue.timeValueSeconds(30L);
+
     @After
     public void ensureExchangesAreReleased() throws Exception {
         for (String node : internalCluster().getNodeNames()) {
@@ -166,7 +168,11 @@ public abstract class AbstractEsqlIntegTestCase extends ESIntegTestCase {
     }
 
     protected final EsqlQueryResponse run(String esqlCommands) {
-        return run(syncEsqlQueryRequest(esqlCommands).pragmas(getPragmas()));
+        return run(esqlCommands, DEFAULT_REQUEST_TIMEOUT);
+    }
+
+    protected final EsqlQueryResponse run(String esqlCommands, TimeValue timeout) {
+        return run(syncEsqlQueryRequest(esqlCommands).pragmas(getPragmas()), timeout);
     }
 
     /** A hook for overriding. */
@@ -175,8 +181,12 @@ public abstract class AbstractEsqlIntegTestCase extends ESIntegTestCase {
     }
 
     public EsqlQueryResponse run(EsqlQueryRequest request) {
+        return run(request, DEFAULT_REQUEST_TIMEOUT);
+    }
+
+    public EsqlQueryResponse run(EsqlQueryRequest request, TimeValue timeout) {
         try {
-            return client().execute(EsqlQueryAction.INSTANCE, request).actionGet(30, TimeUnit.SECONDS);
+            return client().execute(EsqlQueryAction.INSTANCE, request).actionGet(timeout);
         } catch (ElasticsearchTimeoutException e) {
             throw new AssertionError("timeout", e);
         }

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/plugin/RerankIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/plugin/RerankIT.java
@@ -8,6 +8,7 @@
 package org.elasticsearch.xpack.esql.plugin;
 
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.inference.TaskType;
 import org.elasticsearch.xpack.esql.inference.InferenceSettings;
 import org.elasticsearch.xpack.esql.parser.ParsingException;
@@ -147,7 +148,7 @@ public class RerankIT extends InferenceCommandIntegTestCase {
             | KEEP id, _score
             """, testIndexLarge, RERANK_MODEL_ID);
 
-        try (var resp = run(query)) {
+        try (var resp = run(query, TimeValue.timeValueSeconds(60L))) {
             List<List<Object>> values = getValuesList(resp);
             // Should be limited to exactly the default row limit (1000)
             assertThat(values, hasSize(1000));

